TopMovies/Core/Action/MoviesListAction.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  MoviesListAction.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 02.02.2021.
+//
+
+import ReSwift
+
+enum MoviesListAction: Action {
+  case request
+  case downloading
+  case completed(movies: MoviesRelational)
```

Comment:
In theory, it could be fine to use a dictionary in action when you don't care about the order of data.
But if you do, you will always get a random order of movies when you call `relational.values`.
This is why I'd recommend using `[Movies]` in action and transform it in state in an array of keys and in a dictionary.